### PR TITLE
Add index argument to setvar / setglobalvar

### PR DESCRIPTION
**Pull request by LenAnderson**
```
/setvar key=myList ["a","b","c"]
/setvar key=myList index=2 newValue
```
myList is now `["a","b","newValue"]`

If a numeric index is used on a nonexistent variable, the var will be created as an empty array `[]`.
If a string index is used on a nonexistent variable, the var will be created as an empty object `{}`.

### CODE REVIEW

**Comment by Cohee1207** [comment]
Cool idea: /addvar for array types could push an item to array

**Comment by LenAnderson** [comment]
done

File: public/scripts/variables.js [hunk 1]
```diff
@@ -60,13 +81,42 @@ function getGlobalVariable(name, args = {}) {
...
+        extension_settings.variables.global[name] = value;
+    }
     saveSettingsDebounced();
 }
 
 function addLocalVariable(name, value) {
     const currentValue = getLocalVariable(name) || 0;
+    try {
+        const parsedValue = JSON.parse(currentValue);
+        if (typeof parsedValue == 'object' && parsedValue instanceof Array) {
```
Comment (Cohee1207) [inline public/scripts/variables.js]: `Array.isArray(parsedValue)` should do that with less code

Comment (LenAnderson) [reply public/scripts/variables.js]: done

**Comment by LenAnderson** [comment]
let me know if I should split the piping and /# into a separate PR

piping: if a slash command's pipeResult is undefined, the previous pipeResult is passed through

/# serves as a comment in longer ST scripts, it does nothing with its input and returns undefined so that the previous pipeResult is used

```
/getvar myVar |
/# some comments here |
/echo
```
will echo the value of `myVar`

**Comment by Cohee1207** [comment]
> piping: if a slash command's pipeResult is undefined, the previous pipeResult is passed through

I'm not sure about this part. I'll revert it for now, let's think about it later or find potential pros/cons.

**Comment by Cohee1207** [comment]
Ideally, every command should have a return value. If it doesn't return anything specific - that would be an empty string.

**Comment by Cohee1207** [comment]
For comments, it's possible to use the comment macro 
```
/getvar myVar |
{{// some comments here }} |
/echo
```
Any part of the script that is not slash commands is ignored.

**Comment by Cohee1207** [comment]
So, my final verdict - let's make comments a part of the parser somehow, not a separate command

**Pull request merged**